rotkehlchen/db/evmtx.py

```diff
@@ -484,6 +488,14 @@ def delete_transactions(
             tx_hashes=tx_hashes,
             chain_id=chain_id,  # type: ignore[arg-type] # comes from SUPPORTED_EVM_CHAINS
         )
+        dbevents.delete_genesis_tx_events(write_cursor, address)
+        genesis_hash_events_count = dbevents.get_history_events_count(
+            cursor=write_cursor,
+            query_filter=HistoryEventFilterQuery.make(event_identifiers=[GENESIS_HASH]),
+        )
```

Comment:
Here you are are creating a filter and then calling a function just to do a 

```
SELECT COUNT(*) FROM history_events WHERE event_identifier=?
```

I would move it to a simple sql query since you avoid:
1. Creating the filter
2. The extra function call
3. Prepare of the filter
4. Query execution (this is the only thing you do)

The filters are good when there is a complex query involved but here I believe is not needed

Reply:
After thinking a bit you don't need to execute this query. Correct me if I'm wrong but if there are other events related to the genesis tx it means that the `.rowcount` of the previous query `dbevents.delete_genesis_tx_events(write_cursor, address)` (if you move it to a simple sql query) is > 0 because that is the case were nothing was deleted for the genesis tx in `delete_events_by_tx_hash`.

If you add a comment for this would be nice

Reply:
I can change it to a simple query.

`.rowcount` returns the number of rows modified, but doesn't say anything about the remaining events

Reply:
Yes but since you have deleted all the events related to transaction touched by that address and not any other if after deleting the events of this address you still delete some events it means that there is another address having some events related to the geneisis transaction

Reply:
Sorry, I still don't get how can we judge about the remaining number of decoded events in the genesis tx by checking whether current address had any events in the genesis tx.